### PR TITLE
OrbitControls: support panning with left mouse + shift key

### PR DESCRIPTION
**Pull request by leshkovichpvl**
Initially, for the panning you need to use the ctrl/metaKey and this cannot be redefine.
Added the ability to set a custom pan keys, for example shiftKey.

### CODE REVIEW

File: examples/js/controls/OrbitControls.js [hunk 1]
```diff
@@ -679,7 +682,7 @@ THREE.OrbitControls = function ( object, domElement ) {
 
 			case scope.mouseButtons.LEFT:
 
-				if ( event.ctrlKey || event.metaKey ) {
+				if ( scope.panKeys.some( key => event[ key ] ) ) {
```
Comment (Mugen87) [inline examples/js/controls/OrbitControls.js]: Please don't use arrow functions. In general, it's better to define the function once and reuse it in the `some` method call. See the following example code https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

**Review by Mugen87 — CHANGES_REQUESTED** [review]

File: examples/js/controls/OrbitControls.js [hunk 2]
```diff
@@ -423,6 +426,17 @@ THREE.OrbitControls = function ( object, domElement ) {
 
 	}
 
+
+	function isPanEvent( event ) {
+
+		return scope.panKeys.some( function ( key ) {
```
Comment (Mugen87) [inline examples/js/controls/OrbitControls.js]: Sry, for being unclear. But the idea is not to create a function for each call of Array.some.

**Review by Mugen87 — CHANGES_REQUESTED** [review]

File: examples/js/controls/OrbitControls.js (hunk 2 above)
Comment (leshkovichpvl) [reply examples/js/controls/OrbitControls.js]:
```
function isPanEvent( event ) {
	var hasKey = function ( key ) {
		return event[ key ];
	};
	return scope.panKeys.some( hasKey );
}
```
is it so good?

Comment (Mugen87) [reply examples/js/controls/OrbitControls.js]: 👍

Comment (Mugen87) [reply examples/js/controls/OrbitControls.js]: 👍

**Comment by WestLangley** [comment]
From my comment https://github.com/mrdoob/three.js/pull/13972#issuecomment-398236000:

>Note, so as to parallel the Google maps API, I added support for both event.ctrlKey and event.metaKey.

I agree support for a _custom_ pan key is beneficial, but perhaps one key is sufficient:

```javascript
// The panning key
this.panKey = 'ctrlKey';

// ...
if ( event[ scope.panKey ] ) {

**Comment by leshkovichpvl** [comment]
@WestLangley 
or do so:
```this.panKeys = [ 'ctrlKey', 'metaKey', 'shiftKey' ]```
leaving the option to redefine the standard behavior

**Comment by WestLangley** [comment]
It is a judgement call, but my preference is to keep it simple:
```javascript
if ( event.ctrlKey || event.metaKey || event.shiftKey ) {
```
Users are free to modify this code since it is an example.

**Review by Mugen87 — APPROVED** [review]

**Comment by WestLangley** [comment]
@leshkovichpvl Thanks for your suggestion and PR.

One more thing to fix:
```
//    Pan - right mouse, or left mouse + ctrl/meta/shiftKey, or arrow keys / touch: two-finger move
```

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**